src/ape/managers/config.py

```diff
@@ -58,6 +64,15 @@ def _plugin_configs(self) -> Dict[str, ConfigItem]:
         self.version = user_config.pop("version", "")
         self.dependencies = user_config.pop("dependencies", {})
 
+        # Sanitize deployment addresses.
+        deployments = user_config.pop("deployments", {})
+        for ecosystem_name, networks in deployments.items():
+            for network_name, contract_deployments in networks.items():
+                for deployment in [d for d in contract_deployments if "address" in d]:
```

Comment:
I feel like `if "address" in d` should be filtered out by the config item framework somehow (e.g. if neither is in there, the config should fail to parse it raising an error to fix it)

Reply:
Yeah this is probably a remnant for when I had it as a dict

Reply:
Dug in more...

The case where the user leaves off the the address like this:

```
deployments:
  ethereum:
    rinkeby: 
    - contract_type: MyContract

```
 
currently makes it past the parsing stage. Leaving off `if "address" in d` causes a `KeyError` at that part (maybe that is ok?). If we want this to be an error-state, perhaps it is best, like you said, to fail earlier in the config parsing stage.

Reply:
Okay, I made it raise better errors now for this situation!